content/en/docs/concepts/configuration/overview.md

```diff
@@ -73,17 +73,19 @@ A desired state of an object is described by a Deployment, and if changes to tha
 
 ## Container Images
 
-The [imagePullPolicy](/docs/concepts/containers/images/#updating-images) and the tag of the image affect when the [kubelet](/docs/reference/command-line-tools-reference/kubelet/) attempts to pull the specified image.
+The [imagePullPolicy](/docs/concepts/containers/images/#updating-images) and the tag of the image affect when the [kubelet](/docs/reference/command-line-tools-reference/kubelet/) attempts to pull the specified image. For the most reliable results, always set `imagePullPolicy` explicitly, and do not rely on the default behavior.
 
 - `imagePullPolicy: IfNotPresent`: the image is pulled only if it is not already present locally.
 
 - `imagePullPolicy: Always`: every time the kubelet launches a container, the kubelet queries the container image registry to resolve the name to an image digest. If the kubelet has a container image with that exact digest cached locally, the kubelet uses its cached image; otherwise, the kubelet downloads (pulls) the image with the resolved digest, and uses that image to launch the container.
 
-- `imagePullPolicy` is omitted and either the image tag is `:latest` or it is omitted: `Always` is applied.
+- `imagePullPolicy: Never`: the image is assumed to exist locally. No attempt is made to pull the image.
 
-- `imagePullPolicy` is omitted and the image tag is present but not `:latest`: `IfNotPresent` is applied.
+- `imagePullPolicy` is omitted (not recommended):
```

Comment:
I'd write:
```suggestion
- `imagePullPolicy` not set:
   (For the most reliable results, always set `imagePullPolicy` explicitly, and do not rely on the kubelet inferring an image pull policy).
```
and then the similar wording earlier could change to, eg:
```markdown
If reliability is a concern you should ensure that `imagePullPolicy` is set for each container in a Pod to a
value that is appropriate for your workload and the cluster it runs on.
```

Reply:
Hmm, I think I've used the word "reliable" twice in two different ways - one in the SRE sense, the other to mean "we're avoiding surprising behaviour." Perhaps I can change "reliable" to "predictable" for the second sense?

Also, if we don't want to just say "not recommended" in the top-level bullet point, how about adding a third bullet point below the other two saying "As mentioned above, always set `imagePullPolicy` to ensure the most predictable behavior." wdyt?

Reply:
> change "reliable" to "predictable" for the second sense?

:+1:

Rather than telling readers to always set `imagePullPolicy`, the learning outcome I have in mind is that leaving it unset might be OK for quick experiments but isn't the right fit for production workloads. “always” doesn't cover than nuance. I think it's just as important for the reader to see when leaving it out _is_ OK.